theano/tensor/extra_ops.py

```diff
@@ -453,15 +453,11 @@ def perform(self, node, inputs, output_storage):
     def grad(self, inputs, outputs_gradients):
         inputs = inputs[0]
 
-        if inputs.ndim != 1:
-            raise NotImplementedError("Grad is not implemented for inputs with"
-                                      "number of dimension other than 1.")
-
         z = outputs_gradients[0]
 
         def _grad_helper(z):
-            pre = basic.concatenate([[0.], z])
-            app = basic.concatenate([z, [0.]])
+            pre = basic.concatenate([T.zeros(T.sum(z,axis=self.axis,keepdims=True).shape), z],axis=self.axis)
```

Comment:
thanks for the PR. The test theano/tests/test_flake8.py will fail. You need a space after the comma. I'll stop travis as it won't help build this PR and yesterday it was overloaded.

This PR also need test in the file tensor/tests/test_extra_ops.py to test the new fonctionality.